tests/main/generic-unregister/task.yaml

```diff
@@ -0,0 +1,44 @@
+summary: |
+    Ensure that unregistration API works.
+
+# ubuntu-14.04: curl does not have --unix-socket option
+systems: [-ubuntu-core-*, -ubuntu-14.04-*]
+
+prepare: |
+    systemctl stop snapd.service snapd.socket
+    cp /var/lib/snapd/state.json state.json.bak
+    mkdir key
+    cp /var/lib/snapd/device/private-keys-v1/* key
+    systemctl start snapd.service snapd.socket
+
+restore: |
+    systemctl stop snapd.service snapd.socket
+    cp key/* /var/lib/snapd/device/private-keys-v1/
+    cp state.json.bak /var/lib/snapd/state.json
+    systemctl start snapd.service snapd.socket
+
+execute: |
+    #shellcheck source=tests/lib/core-config.sh
+    . "$TESTSLIB"/core-config.sh
+
+    wait_for_device_initialized_change
+
+    snap model --assertion | MATCH "series: 16"
+
+    if snap model --verbose | NOMATCH "brand-id:\s* generic" ; then
+       echo "Not a generic model. Skipping."
+       exit 0
+    fi
+
+    curl --data '{"action":"forget","no-registration-until-reboot":true}' --unix-socket /run/snapd.socket http://localhost/v2/model/serial
```

Comment:
would be kind of nice to have the `no-registration-until-reboot: false` case tested as well where we unregister then restart snapd and see that it has re-registered itself automatically

Reply:
I can but probably as follow up as writing this is a pain and our main use case ATM is with the flag set